### PR TITLE
feat(Hero): create multiple hero components for new DS

### LINKED ISSUE #10753: Hero
Create a new component for the new Hero component specified in [the DS](https://www.figma.com/file/NrNxGjBL0Yl1PrNrOT8G2B/ethereum.org-Design-System?node-id=2486%3A48375).

image

Depends on #11140

### PR DESCRIPTION
**Pull request by TylerAPfledderer**
## Description

Adds a new set of Hero components per DS.

- Homepage Hero and story
- Hub Hero and story
- Content Hero and story
- MDX hero and story
- Content Hero with Big Number stats not included (to be added in a future PR)
- New hero illustration implemented for the Home Page and Learning Hub only

## Related Issue
Closes #10753 

Part of Implementation of DS v1 #9548

### CODE REVIEW

**Comment by TylerAPfledderer** [comment]
@nloureiro any update with Hub hero illustrations?

**Comment by nloureiro** [comment]
> @nloureiro any update with Hub hero illustrations?

for the new illustration? we might still be one week from getting the first one.

are these new illustrations blocking you from any developments?

**Comment by TylerAPfledderer** [comment]
@nloureiro not a blocker here. I was thinking of just keeping this in draft until they become available. Also keeping it indraft because there is also another primitive component from the DS still missing for this (Big Numbers component for content hero)

**Comment by nloureiro** [comment]
> 

Oh, the big numbers. I'll be sure to look into it. 

regarding the rest, I wanted to have everything ready and tested before we have the illustrations. 
Can this be removed from the draft and have a developer preview to test on different browsers and resolutions?

I sense we need some design adjustments, and I wanted to do it on the browser.

Thank you for your help :)

**Comment by TylerAPfledderer** [comment]
@nloureiro I went ahead and switched to "Ready for Review" :)

**Comment by nloureiro** [comment]
> @nloureiro I went ahead and switched to "Ready for Review" :)
> 
> https://63b7ea99632763723c7f4d6b-dgtmjvennt.chromatic.com/?path=/story/organisms-layouts-hero--home-hero

Looks great! I approve it on Chromatic, I wonder if that leads to an approval here on the PR itself.

One last thing that I wanted to test is on an actual page.
 Can this be on the learning hub page on the developer preview?

Thank you

**Comment by TylerAPfledderer** [comment]
> > @nloureiro I went ahead and switched to "Ready for Review" :)
> > 
> > https://63b7ea99632763723c7f4d6b-dgtmjvennt.chromatic.com/?path=/story/organisms-layouts-hero--home-hero
> 
> Looks great! I approve it on Chromatic, I wonder if that leads to an approval here on the PR itself.
> 
> One last thing that I wanted to test is on an actual page.
>  Can this be on the learning hub page on the developer preview?
> 
> Thank you
> 
> 
> 
> 

@nloureiro the approval you made was through the "UI Review" check on here, which did update. "UI Tests" remains pending.

I can update the learning page using the [Eth2 Illustration](https://github.com/ethereum/ethereum-org-website/blob/0b4f4e952c68ea25e50bb2112b9bd80392336db4/src/assets/ethereum-eth2.png)

**Comment by nloureiro** [comment]
> > > @nloureiro I went ahead and switched to "Ready for Review" :)
> > > https://63b7ea99632763723c7f4d6b-dgtmjvennt.chromatic.com/?path=/story/organisms-layouts-hero--home-hero
> > 
> > 
> > Looks great! I approve it on Chromatic, I wonder if that leads to an approval here on the PR itself.
> > One last thing that I wanted to test is on an actual page.
> > Can this be on the learning hub page on the developer preview?
> > Thank you
> 
> @nloureiro the approval you made was through the "UI Review" check on here, which did update. "UI Tests" remains pending.
> 
> I can update the learning page using the [Eth2 Illustration](https://github.com/ethereum/ethereum-org-website/blob/0b4f4e952c68ea25e50bb2112b9bd80392336db4/src/assets/ethereum-eth2.png)

got, just approve it

Yes, great, we can use the Eth2 illustrations for testing purposes. Thank you!!!

**Comment by TylerAPfledderer** [comment]
@nloureiro updated the [Learning Hub Page](https://ethereumorgwebsitedev01-tylerapfleddererethereumo12683.gatsbyjs.io/en/learn/)

**Review by pettinarip — COMMENTED** [review]
Good job @TylerAPfledderer 

Is this ready for review? as I see in the comments that you changed it bc you needed the preview deploy for testing purposes. You are also probably waiting for #11133 to be merged (will be soon).

**Comment by TylerAPfledderer** [comment]
> Good job @TylerAPfledderer 
> 
> Is this ready for review? as I see in the comments that you changed it bc you needed the preview deploy for testing purposes. You are also probably waiting for #11133 to be merged (will be soon).

There is GatsbyImage as well as getting the Big Numbers component online too.

I don't see the latter being a blocker if that can be added in a future PR, if currently these heroes are only to be tested in preview and Storybook and not yet deployed to prod.

**Comment by pettinarip** [comment]
> > Good job @TylerAPfledderer
> > Is this ready for review? as I see in the comments that you changed it bc you needed the preview deploy for testing purposes. You are also probably waiting for #11133 to be merged (will be soon).
> 
> There is GatsbyImage as well as getting the Big Numbers component online too.
> 
> I don't see the latter being a blocker if that can be added in a future PR, if currently these heroes are only to be tested in preview and Storybook and not yet deployed to prod.

Got it. FYI #11133 is merged now.

File: src/components/Hero/MdxHero/index.tsx
Comment (pettinarip) [inline src/components/Hero/MdxHero/index.tsx:7]: To be consistent with the rest
```suggestion
  title: string
```

File: src/components/Hero/ContentHero/index.tsx
Comment (pettinarip) [inline src/components/Hero/ContentHero/index.tsx:9]: To be consistent with the MdxHero api, I'd call this `breadcrumb` and accept any breadcrumb prop.

```tsx
// and then just
<Breadcrumbs {...breadcrumbs} />
```

**Review by pettinarip — COMMENTED** [review]
Beautiful!

Looking great @TylerAPfledderer.

I think we should also add a `components/Hero/index.ts` file to export all the nested hero components, as we are doing in other places, to be consistent and have better autocomplete dx when importing.

File: docs/ds-implementation.md
Comment (TylerAPfledderer) [reply docs/ds-implementation.md:23]: @pettinarip Darnit. Every time I try to save the changes locally, it wants to fix the formatting. -_-

Comment (pettinarip) [reply docs/ds-implementation.md:23]: Yea, I think is because it is coming as a new/modified file when you update the branch with dev (that is caught by prettier). Shouldn't happen in the upcoming PRs you make.

File: src/components/Hero/index.ts
Comment (pettinarip) [reply src/components/Hero/index.ts:1]: Yea, good point, I think that consistency in how we export from entry points would be an improvement. If you can do that in the other PRs as well, that would be awesome 👍🏼 thanks

File: src/components/Hero/ContentHero/index.tsx
Comment (TylerAPfledderer) [reply src/components/Hero/ContentHero/index.tsx:36]: I'd work this out in a separate PR, of course 😅

File: src/components/Hero/HubHero/index.tsx
Comment (TylerAPfledderer) [inline src/components/Hero/HubHero/index.tsx:61]: @pettinarip I brought this over from current usage in the hub pages. Should this really be a heading? And more importantly, should this be an `h1` versus the other heading below?

image
Here, `Learn Hub` is the h1, whereas the more prominent `Learn About Ethereum` is an h2.

I see this as semantics in this area of the page, though there are two things to point out as far as a screen reader is concerned:
1. `Learn Hub` is already used in the `title` tag for this page, so when the user enters this page, the screen reader will already declare this before the user begins navigating. So from that standpoint, giving the `Learn Hub` heading an `h1` is redundant. Having it as the `Text` component using `span` semantically would be better: along with the redundancy it is not labeling a section (including what I describe in # 2) and the styling lends to this usage.
2.  `Learn About Ethereum` is really the title that describes the page. And `h1` as the heading that should only be used once per page fits the bill here.

File: src/components/Hero/utils.ts
Comment (TylerAPfledderer) [inline src/components/Hero/utils.ts:9]: Want to point this out to make sure intention is clear.
This typing for `buttons`, where...
- The prop itself is `optional`
- It specifies a tuple of the type `CTAParentProps`
- The second "element" of the tuple is `optional`

means that 0 - 2 buttons can be rendered but no more than two, as the design calls for only up to two. Else a type error will throw.

File: src/components/Hero/utils.ts
Comment (TylerAPfledderer) [inline src/components/Hero/utils.ts:9]: Want to point this out to make sure intention is clear.
This typing for `buttons`, where...
- The prop itself is `optional`
- It specifies a tuple of the type `CTAParentProps`
- The second "element" of the tuple is `optional`

means that 0 - 2 buttons can be rendered but no more than two, as the design calls for only up to two. Else a type error will throw.

Comment (pettinarip) [reply src/components/Hero/utils.ts:9]: Perfecto! clear enough, thanks for thinking about that!

File: src/components/Hero/HubHero/index.tsx
Comment (pettinarip) [reply src/components/Hero/HubHero/index.tsx:59]: Thanks for sharing these observations and thoughts.

As far as I remember, the reasons behind these semantics were SEO and the way Algolia indexes the content.

A clear example is in the [layer 2 page](https://ethereum.org/en/layer-2/):
image

The idea is to have the "formal" and more concrete heading title at the top of the ranking. The current h1, in this case, "Layer 2". That one will appear in the search engines, and that title should not require context to be understood.

The second title, in this case, "Ethereum for everyone" is the more "friendly" subtitle that needs the context of the page to be understood but we don't want that to be the first thing to pop up in search results. That's why it has an h2.

Hope that clarifies a bit the logic here.

cc @wackerow I'm sure you could confirm or correct me if I'm wrong.

======

Then, in terms of SR, you are right. Wondering if we could somehow fix that 🤔 without losing the previously mentioned things.

Not the most ideal solution but I guess that we could have a `sr-only` class for these cases, no?

**Comment by pettinarip** [comment]
@TylerAPfledderer I've updated the PR fixing the conflict with the ds guide

File: src/components/Hero/HubHero/index.tsx
Comment (TylerAPfledderer) [reply src/components/Hero/HubHero/index.tsx:59]: Oh! For my awareness at least I will take a look at SEO conventions related to this.

Ok, I understand now with the h2 usage!

> Not the most ideal solution but I guess that we could have a sr-only class for these cases, no?

That would mean the text is not visible on the page and the screen reader will read it out. Not the case here. There would be the screen reader ignoring the text, but you don't want it to ignore a heading.

I'll revisit this along with the SEO. I will take no action on the code for now because of the explanation with the large subtitle.

Comment (pettinarip) [reply src/components/Hero/HubHero/index.tsx:59]: Ah right! yea, `sr-only` is not going to work in this case.

**Comment by TylerAPfledderer** [comment]
> Hello there! few comments  on design:
> 
> 1.  The line height for headline should have less space between the headline lines than between the headline and the paragraph below it 
> 2. The mobile design needs some love for the spacing around button
> 3. Do we need to have the small grey text "learn hub" there? i dont think so if its only for the hub pages it serves no purpose

@konopkja 

As to #1, this will be reviewed

As to #2, I think the container for the content below the hero should receive the padding or margin to gain more space.

I was wondering about implementing these heroes right away into prod, versus opening a separate PR to address layout tweaks that would be needed. The example you gave is one of those, and the home page needs to be readdressed to allow for the Hero image to span the entire viewport (the current wrapping container does not allow this)

**Comment by nloureiro** [comment]
> Hello there! few comments on design:
> 
> 1. The line height for headline should have less space between the headline lines than between the headline and the paragraph below it
> 2. The mobile design needs some love for the spacing around button
> 3. Do we need to have the small grey text "learn hub" there? i dont think so if its only for the hub pages it serves no purpose
> 
> Screenshot 2023-10-03 at 19 59 49
> 
> image



good points you raised:

1. The implementation follows the design, I think the heading line-height matched the spacing below. for it's not a big issue, let me know if you feel like it's a show stopper that needs a figma review
Screen Shot 2023-10-03 05 26 35 PM


2. yes, it's a bug we need more space here for sure.
@TylerAPfledderer is this spacing given by the hero? or is the content? > my question is more should the hero have the 4 rem spacing in it?

Screen Shot 2023-10-03 05 21 55 PM


3. we've never disscuss this in detail, have we? for now it's the H1 of the page and it helps for SEO puporses but I personaly like it, feels like an intro to the page and realises the bigger text to be more creative.
I would leave as it it for this PR, unless you feel otherwise.

**Comment by nloureiro** [comment]
> > Hello there! few comments  on design:
> > 
> > 1. The line height for headline should have less space between the headline lines than between the headline and the paragraph below it
> > 2. The mobile design needs some love for the spacing around button
> > 3. Do we need to have the small grey text "learn hub" there? i dont think so if its only for the hub pages it serves no purpose
> 
> @konopkja
> 
> As to #1, this will be reviewed
> 
> As to #2, I think the container for the content below the hero should receive the padding or margin to gain more space.
> 
> I was wondering about implementing these heroes right away into prod, versus opening a separate PR to address layout tweaks that would be needed. The example you gave is one of those, and the home page needs to be readdressed to allow for the Hero image to span the entire viewport (the current wrapping container does not allow this)


are you thinking to start the implementation of this:
Screen Shot 2023-10-03 05 46 37 PM

🙌

**Comment by konopkja** [comment]
@nloureiro  1 and 3 are definitely not show-stoppers, but i think they should be addressed instead of accepted and ignored. If this PR isnt a goot place for it, we should address it on one of our design syncs.

**Comment by nloureiro** [comment]
> @nloureiro 1 and 3 are definitely not show-stoppers, but i think they should be addressed instead of accepted and ignored. If this PR isnt a goot place for it, we should address it on one of our design syncs.

Agree. let move this PR forward and circle back on #1 and #2. 

Let's go!

**Comment by TylerAPfledderer** [comment]
@nloureiro for spacing between sections, I would have a parent component handle the spacing, and not make the hero be responsible.

So have a wrapper define spacing between the hero and the content below it.

@pettinarip thoughts on trying to implement the heroes like the home hero in this PR, even if it requires making layout adjustments for the whole page? I'm sure if it just needs minor tweaks until I give it a go.

**Comment by nloureiro** [comment]
> @nloureiro for spacing between sections, I would have a parent component handle the spacing, and not make the hero be responsible.
> 
> So have a wrapper define spacing between the hero and the content below it.

ok, that makes sense. 

would you fix the template on this PR? is that it?

**Comment by TylerAPfledderer** [comment]
I could. Along with the home page tweaking and other possible templates that need adjustments, I was concerned that it might bloat the PR. But I can go ahead and take a look!

**Comment by TylerAPfledderer** [comment]
Also, I checked the line-height for the heading. This is a bug(?). The tokens used for the breakpoints of the `2xl` heading size are implemented backwards (the larger line-height of the two used is being rendered at the smaller breakpoint instead of the larger breakpoint)

I'm going to create a separate PR to address this, because it looks like they are backwards for other heading sizes in the [component theme file](https://github.com/ethereum/ethereum-org-website/blob/a5c06cdff458f4ad81ab14fe3ddd01158847ba44/src/%40chakra-ui/gatsby-plugin/components/Heading.ts#L6-L15)

**Comment by nloureiro** [comment]
hey @TylerAPfledderer 

we have the final image for the learning hub

we can prepare the final version for realise 🚀
Learn_hero

**Comment by TylerAPfledderer** [comment]
> hey @TylerAPfledderer 
> 
> we have the final image for the learning hub
> 
> we can prepare the final version for realise 🚀
> Learn_hero
> 
> 

@nloureiro 
To clarify, I wouldn't put the hub hero into prod because I have to account for the other hub pages with new images...

Or just keep the updated learning hub page only for prod with the new hero component and this new image?

**Comment by nloureiro** [comment]
> > hey @TylerAPfledderer
> > we have the final image for the learning hub
> > we can prepare the final version for realise 🚀
> > Learn_hero
> 
> @nloureiro To clarify, I wouldn't put the hub hero into prod because I have to account for the other hub pages with new images...
> 
> Or just keep the updated learning hub page only for prod with the new hero component and this new image?


I'm not sure if I understand, but what I have in mind is to publish just this hero now.
Whenever we have new illustrations we will change the hero to "that" one page on a seperate PR

does it make sense?

**Comment by pettinarip** [comment]
> Also, I checked the line-height for the heading. This is a bug(?). The tokens used for the breakpoints of the `2xl` heading size are implemented backwards (the larger line-height of the two used is being rendered at the smaller breakpoint instead of the larger breakpoint)
> 
> I'm going to create a separate PR to address this, because it looks like they are backwards for other heading sizes in the [component theme file](https://github.com/ethereum/ethereum-org-website/blob/a5c06cdff458f4ad81ab14fe3ddd01158847ba44/src/%40chakra-ui/gatsby-plugin/components/Heading.ts#L6-L15)
> 
> Edit: See #11363

Merged

**Comment by nloureiro** [comment]
@TylerAPfledderer  did you fix the spacing bug on mobile?

Screen Shot 2023-10-18 03 01 59 PM

I think was the only blocker a few weeks ago.

**Review by pettinarip — COMMENTED** [review]
I took another look at this and everything looks good 👍🏼 

Once we add that new image I think we are ready to go!

**Comment by TylerAPfledderer** [comment]
> @TylerAPfledderer did you fix the spacing bug on mobile?
> 
> Screen Shot 2023-10-18 03 01 59 PM
> 
> I think was the only blocker a few weeks ago.

Yep, I can go ahead and make that adjustment here!

**Comment by TylerAPfledderer** [comment]
@nloureiro @pettinarip added the newest image for the learning hub and applied the spacing beneath the hero for that page.
https://ethereumorgwebsitedev01-tylerapfleddererethereumo12683.gatsbyjs.io/en/learn/

After updating the branch with the latest `dev` the line-height issues should also be fixed via #11363.

Also with the spacing issue in mobile: since it also showing in current prod, I would like to know if the spacing I applied is enough; I could not find a reference to work with.

**Comment by TylerAPfledderer** [comment]
@nloureiro I added in that updated image file. I do see an improvement.

Currently we have a graphQL query that maxes the image to 1504px wide with max quality for a constrained layout.

**Comment by nloureiro** [comment]
> @nloureiro I added in that updated image file. I do see an improvement.
> 
> Currently we have a graphQL query that maxes the image to 1504px wide with max quality for a constrained layout.

yes, indeed it looks better!
oh, I did not know about the 1504px > why this size?

and, are adopting to better resolution screens, like 2x and 3x, using this script?

**Comment by TylerAPfledderer** [comment]
> > @nloureiro I added in that updated image file. I do see an improvement.
> > Currently we have a graphQL query that maxes the image to 1504px wide with max quality for a constrained layout.
> 
> yes, indeed it looks better! oh, I did not know about the 1504px > why this size?
> 
> and, are adopting to better resolution screens, like 2x and 3x, using this script?

This was something in place before I touched anything.

`1504px` is the max width of the container of the page content at large screens, since the image stretches the full width of its container. Gatsby will only generate a image file no larger than `1504px` from the original, as well as multiple images down a set of breakpoints so the page loads an appropriate size image based on the screen size. This keeps load times down with file sizes.

Aspect ratio is maintained. I don't think it is altering the image like 3x or 2x resolution.

**Review by pettinarip — APPROVED** [review]
🚢 🚢 🚢 !!

**Pull request merged**
